### ISSUE TITLE
"index out of range" cause ExecuteResponse.Results were empty.

### ISSUE THREAD

**Issue by funny-falcon** [Boostport/avatica#48]
Got an error:
```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/Boostport/avatica.(*stmt).exec(0xc4201e80f0, 0x8f2ee0, 0xc4200160a0, 0xc4201e81e0, 0x2, 0x2, 0x0, 0x0, 0x709460, 0x738800)
        /home/yura/go/src/github.com/Boostport/avatica/statement.go:81 +0x2c3
github.com/Boostport/avatica.(*stmt).ExecContext(0xc4201e80f0, 0x8f2ee0, 0xc4200160a0, 0xc4201e8140, 0x2, 0x2, 0xc400000008, 0xc420399d00, 0x0, 0x4c8b06)
        /home/yura/go/src/github.com/Boostport/avatica/statement_go18.go:19 +0x117
database/sql.ctxDriverStmtExec(0x8f2ee0, 0xc4200160a0, 0x8f3120, 0xc4201e80f0, 0xc4201e8140, 0x2, 0x2, 0x2, 0x2, 0x0, ...)
        /snap/go/current/src/database/sql/ctxutil.go:66 +0x22e
database/sql.resultFromStatement(0x8f2ee0, 0xc4200160a0, 0x8f1960, 0xc420110120, 0xc420281b08, 0xc420281eb0, 0x2, 0x2, 0x0, 0x0, ...)
        /snap/go/current/src/database/sql/sql.go:2109 +0x184
database/sql.(*DB).execDC(0xc42008a960, 0x8f2ee0, 0xc4200160a0, 0xc420134070, 0xc420281c60, 0x77b0de, 0xd5, 0xc420281eb0, 0x2, 0x2, ...)
        /snap/go/current/src/database/sql/sql.go:1272 +0x2c7
database/sql.(*DB).exec(0xc42008a960, 0x8f2ee0, 0xc4200160a0, 0x77b0de, 0xd5, 0xc420281eb0, 0x2, 0x2, 0xc420016101, 0x6eb580, ...)
```
Looks like Results were empty in this row:
```go
    // Currently there is only 1 ResultSet per response
    changed := int64(res.(*message.ExecuteResponse).Results[0].UpdateCount)                                                                                                                                                         
```
Sorry, I don't know how reliably reproduce it. I was copying large table to other table (using `upsert into t2 select some from t1`), and it happens at night. Probably it is due to whole cluster instability (other clients died with "cluster is out of memory" probably at same time).

**Comment by F21** [comment]
Yes, it looks like `Result` is empty. Since `ExecuteResponse.Results` is a protobuf message and protobuf fields are option, I think we should add a check for it. Did the data insert properly? I am assuming it didn't (due to out of memory) so I think it's safe to return an error if `ExecuteResponse.Result` has a length of less than 1.

**Comment by F21** [comment]
Released as `v2.2.4`
